deeppavlov/core/layers/attention.py

```diff
@@ -0,0 +1,107 @@
+# Copyright 2017 Neural Networks and Deep Learning lab, MIPT
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Optional
+
+import tensorflow as tf
+
+
+class MultiHeadSelfAttention(tf.keras.layers.Layer):
+    """
+    Simplified version of multi-head attention. It uses just an input mask (if provided) to calculate attention mask.
+
+    Args:
+          hidden_size: hidden size of the Transformer (d_model)
+          num_heads: number of attention heads
+          neg_inf: some number close to -infinity to add to logits in order to mask corresponding values
+          attention_probs_dropout_prob: dropout probability of the attention probabilities.
+          trainable: whether the layer's variables should be trainable
+          **kwargs: keyword arguments for base Layer class
+    """
+    def __init__(self,
+                 hidden_size: int = 768,
+                 num_heads: int = 12,
+                 neg_inf: float = -10000.0,
```

Comment:
I was forced to go to the NumPy docs to discover what you wrote above...
Is it necessary to do for other users?
I made this parameter configurable just because I saw -1e9 in one of the implementations of MultiHeadAttention... But after softmax, you will not see any differences between these two options anyway (I checked).
One of the options is to hardcode this as it is done in the official implementation. What do you think?

Reply:
This parameter is useless, so it just spoils the interface.
Yes, we can do hardcode, I just think that the numpy provided one will be better. But I can live with -1e9.